### PR TITLE
Docs: update versions on config file page

### PR DESCRIPTION
**Pull request by humitos**
- Update all the versions used in the examples to the latest ones supported by Read the Docs.
- Remove deprecated keys from examples, like `python.version: "3.7"`

This way we avoid new users copy and pasting examples with outdated versions.

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9838.org.readthedocs.build/en/9838/

- Developer's documentation (`dev`): https://dev--9838.org.readthedocs.build/en/9838/

### CODE REVIEW

**Comment by benjaoming** [comment]
Other files found:

```
docs/user/faq.rst
260:      - python=3.9

docs/user/tutorial/index.rst
325:       python: "3.8"
341:You will notice that one of the lines contains ``python3.8``:
345:.. figure:: /_static/images/tutorial/build-python3.8.png
388:       python: "3.8"

docs/user/environment-variables.rst
106:       python: 3.10

docs/user/guides/conda.rst
51:      - python=3.8
59:   whereas ``python=3.8`` (single ``=``) will fetch the latest ``3.8.*`` version,
98:      - python=3.8
157:      - python=3.8
194:      - python=3.8

```

**Comment by benjaoming** [comment]
> This way we avoid new users copy and pasting examples with outdated versions.

:+1: 

(also we avoid users thinking that we somehow don't support newer versions of Python!)

**Review by benjaoming — APPROVED** [review]
I noted some other files with Python versions in examples. I don't want to scope creep this, as the current update is already a much-needed improvement.

We could need replacement variables like `|example_python_version|`, `|example_sphinx_version|` etc... although they might a) not work in code blocks and b) might not work in _any_ context.

File: docs/user/guides/reproducible-builds.rst [hunk 1]
```diff
@@ -197,50 +197,47 @@ and it generates a ``requirements.txt`` file with the full set of transitive dep
...
       :caption: docs/requirements.txt
 
-      # This file is autogenerated by pip-compile with python 3.7
-      # To update, run:
       #
-      #    pip-compile docs.in
+      # This file is autogenerated by pip-compile with Python 3.10
+      # by the following command:
+      #
+      #    pip-compile test.in
```
Comment (humitos) [inline docs/user/guides/reproducible-builds.rst:209]:
```suggestion
      #    pip-compile docs.in
```

File: docs/user/guides/reproducible-builds.rst [hunk 2]
```diff
@@ -197,50 +197,47 @@ and it generates a ``requirements.txt`` file with the full set of transitive dep
...
           # via babel
-      requests==2.27.1
+      requests==2.28.1
           # via sphinx
       snowballstemmer==2.2.0
           # via sphinx
-      sphinx==4.4.0
-          # via -r docs.in
+      sphinx==5.3.0
+          # via -r test.in
```
Comment (humitos) [inline docs/user/guides/reproducible-builds.rst:240]:
```suggestion
          # via -r docs.in
```

**Comment by humitos** [comment]
Failing test is trim API 😄

**Pull request merged**